lib/spanner/command/request.ex

```diff
@@ -48,10 +48,15 @@ defmodule Spanner.Command.Request do
 
   defp read_config(request, config_path) do
     [bundle, _cmd] = String.split(request.command, ":", parts: 2)
-    cmd_config_file = Path.join([config_path, bundle, Config.dynamic_file_name()])
-    case File.exists?(cmd_config_file) do
+    cmd_config_dir = Path.join([config_path, bundle])
+
+    case File.dir?(cmd_config_dir) do
       true ->
-        Config.Parser.read_from_file(cmd_config_file)
+        with {:ok, files} <- File.ls(cmd_config_dir),
+             cmd_config_file when is_binary(cmd_config_file) <- Enum.find(files, {:ok, ""}, &Spanner.Config.config_file?/1) do
```

Comment:
This bit mayhaps could be clearer. The file is optional. If it isn't there we just carry on. So we aren't actually guaranteed that we will find a config file. That's what the guard for binary is for. Then we return `{:ok, ""}` and move on.

That said, it may not be a terrible idea to log a warning that the dir for dynamic config is available but there wasn't a file found. In which case returning `:not_found` instead of `{:ok, ""}` would probably be a good idea.